taglibs/async/await-tag.js

```diff
@@ -23,6 +23,10 @@ function promiseToCallback(promise, callback, thisObj) {
             finalPromise = finalPromise.fail(function(err) {
                 callback(err);
             });
+        } else {
+            finalPromise = finalPromise.then(void 0, function(err) {
```

Comment:
As of IE9, the global `undefined` is immutable (you can still create a local variable named `undefined`).  

`void 0 === undefined` is safer as `void` is a reserved word and cannot be overridden.  I do think `undefined`/`null` reads better though.

Interestingly, the native `Promise` implementation doesn't care what you pass.  If it's a function it'll call it, otherwise not.  `promise.then('foo', console.log)` works perfectly fine.